jobspec/parse_service.go

```diff
@@ -228,6 +228,54 @@ func parseSidecarService(o *ast.ObjectItem) (*api.ConsulSidecarService, error) {
 	return &sidecar, nil
 }
 
+func parseSidecarTask(item *ast.ObjectItem) (*api.SidecarTask, error) {
+	// We need this later
+	var listVal *ast.ObjectList
+	if ot, ok := item.Val.(*ast.ObjectType); ok {
+		listVal = ot.List
+	} else {
+		return nil, fmt.Errorf("should be an object")
+	}
+
+	// Check for invalid keys
+	valid := []string{
```

Comment:
It might make it easier for reviewers in the future when we add fields if the order of fields here mirrored the order in the struct?